code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -0,0 +1,161 @@
+
+/obj/item/weapon/gun/projectile/sniper_rifle
+	name = "sniper rifle"
+	desc = "the kind of gun that will leave you crying for mummy before you even realise your leg's missing"
+	icon_state = "moistnugget"
+	item_state = "moistnugget"
+	recoil = 2
+	heavy_weapon = 1
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds
+	fire_delay = 40
+	origin_tech = "combat=8"
+	var/zoomed = FALSE
+	var/zoom_amt = 7
+	var/datum/action/sniper_zoom/azoom
+
+/obj/item/weapon/gun/projectile/sniper_rifle/syndicate
+	name = "syndicate sniper rifle"
+	desc = "syndicate flavoured sniper rifle, it packs quite a punch, a punch to your face"
+	pin = /obj/item/device/firing_pin/implant/pindicate
+	origin_tech = "combat=8;syndicate=4"
+
+/obj/item/weapon/gun/projectile/sniper_rifle/New()
+	..()
+	azoom = new()
+	azoom.rifle = src
+
+/obj/item/weapon/gun/projectile/sniper_rifle/dropped(mob/living/user)
+	azoom.Remove(user)
+
+/obj/item/weapon/gun/projectile/sniper_rifle/pickup(mob/living/user)
+	azoom.Grant(user)
+
+/obj/item/weapon/gun/projectile/sniper_rifle/proc/zoom(mob/living/user)
+	if(!user || !user.client)
+		return
+
+	zoomed = !zoomed
+	if(zoomed)
+		var/_x = 0
+		var/_y = 0
+		switch(user.dir)
+			if(NORTH)
+				_y = zoom_amt
+			if(EAST)
+				_x = zoom_amt
+			if(SOUTH)
+				_y = -zoom_amt
+			if(WEST)
+				_x = -zoom_amt
+
+		user.client.pixel_x = world.icon_size*_x
+		user.client.pixel_y = world.icon_size*_y
+	else
+		user.client.pixel_x = 0
+		user.client.pixel_y = 0
+
+/obj/item/weapon/gun/projectile/sniper_rifle/can_shoot()
+	if(..())
+		if(zoomed)
+			return 1
+	return 0
```

Comment:
This apparently didnt even work anyway, and I've grown distant from it.

Noscopes it is.